capitalone/rubicon
# revisit examples

closes: #73

---

## What

**general updates/enhancements**

I came across a handful of small things that needed to be fixed to get all the examples running again:

  * updated the docs makefile to copy the examples to the appropriate subdirectories
    * this fixes a bug where the "view on GitHub" links aren't currently working
  * ran `pre-commit update` to get libraries to latest versions
  * removed `match_versions.sh` and moved the environment file to the repo's root
  * added CSS to make dataframes look good in the docs' dark mode
  * fixed a bug in the UI model where memory persistence with a default `root_dir` fails
    * I think it's another bug that the default `root_dir` ended up as `None` rather than `./`
  * fixed a bug where the async library needed `detail=True` added to a `fsspec` call

specific example notebook changes:

  * removed the auto-executing from all the examples in favor of just committing them executed
    * theres value in the outputs, so why not show them
  * quick look
    * made it a notebook and got rid of the RST version of it
      * I know I was the one who wanted them to be separate in the first place - that was dumb lol
    * removed some things that were more in-depth than a quick look
      * S3 sync, async, multiprocessing
  * collab and review
    * I just got rid of this one - I think we could really use a whole milestone around making the Intake stuff feel like a part of Rubicon rather than just a throwaway add on
  * logging basics
    * split the actual logging part into more than one cell
    * got rid of the dashboard stuff - not logging basics
  * logging concurrently
    * at some point this one stopped working because `multiprocessing` doesn't play nicely with iPython
    * I had to move the code into a script to get it to actually execute

## How to Test
  * build the new docs off this branch and check out the examples
  * run through all the example notebooks
